linera-core/src/client.rs

```diff
@@ -1589,11 +1606,62 @@ where
             operations: Vec::new(),
             previous_block_hash: self.block_hash,
             height: self.next_block_height,
-            authenticated_signer: None,
+            authenticated_signer: owner,
             timestamp,
         };
-        let (_, response) = self.node_client.stage_block_execution(block).await?;
-        Ok(response.info.system_balance)
+        match self
+            .stage_block_execution_and_discard_failing_messages(block)
+            .await
+        {
+            Ok((_, response)) => Ok((
+                response.info.system_balance,
+                response.info.requested_system_balance,
+            )),
+            Err(_) => {
+                // It seems like we can't even stage the execution of an empty block.
```

Comment:
But why? Shouldn't we make that work? Feels like empty blocks should be allowed.

Reply:
It may fail because of the block fees.